### DISCUSSION TITLE
How to use PQC between HTTP client (in go language) and server(in python language)

### DISCUSSION THREAD

**Discussion by ranjankumar23** [open-quantum-safe/liboqs#1532] [Q&A]
I have one application in which i am using HTTP client (Go) and HTTP server (python). Now want to enable PQC for this HTTPS.

What are the options i have
1. convert complete server and client code to cpp and use PQC enabled openssl.
- It is challenge as existing code is big and tested.
3. update go and python HTTP part to use PQC. 
- Don't know how to do this.

Please share me your inputs.

**Comment by ranjankumar23** [comment]
@sebastinas how to use PQC in HTTP client and Server?

**Comment by baentsch** [comment]
> update go and python HTTP part to use PQC.

I have no experience with go, but for python check out [this Dockerfile](https://github.com/open-quantum-safe/liboqs-python/blob/main/docker/Dockerfile): Even though it's unsupported OpenSSL111 code, upgrading to OpenSSL3 should be straightforward if this is what you're looking for.

**Reply by ranjankumar23** [reply]
Michael @baentsch Thanks for your reply. In this liboqs-python directory i could not find how to set 'kyber768' as KEM.

For example we call this function in c/cpp.
int ret =  SSL_CTX_set1_groups_list(ctx, "kyber768");

But in python SSLCOntext i did not find any function to set group. Please check from this as starting point.

**Reply by baentsch** [reply]
> in python SSLCOntext i did not find any function to set group. Please check from this as starting point.

Then this is a (lack in the) Python API -- but no OQS issue. You may want to consider setting (PQ-) Groups alternatively via the `openssl.cnf` mechanism undoubtedly underlying the Python integration. See e.g., [here](https://github.com/open-quantum-safe/oqs-demos/blob/main/epiphany/openssl-client.cnf) for an example how to do this. Just make sure the file is getting used (e.g., by setting OPENSSL_CONFIG env var).
